src/Interpreters/Cache/QueryCache.cpp

```diff
@@ -115,6 +116,13 @@ ASTPtr removeQueryCacheSettings(ASTPtr ast)
     return transformed_ast;
 }
 
+String queryStringFromAST(ASTPtr ast)
+{
+    WriteBufferFromOwnString buf;
+    formatAST(*ast, buf, /*hilite*/ false, /*one_line*/ true, /*show_secrets*/ false);
+    return buf.str();
+}
```

Comment:
why not use `serializeAST(ast)` ?https://github.com/ClickHouse/ClickHouse/blob/4a3f5234af8f217b4187648212a428bf3d8f9d2d/src/Parsers/formatAST.cpp#L15-L19

Reply:
I'll try but it looks like I need to do some cleanup first in order to do this.

Reply:
The problem is that the query cache needs to serialize the AST as text without secrets (passwords etc.).
- I could either add a `show_secrets` (default) parameter to `serializeAST()` but that would be a query-cache-specific parameter in a general-purpose call.
- Alternatively, I could provide an overload that takes a `IAST::FormatSetting` but that would again be a query-cache-specific thing.

I guess it's the best to keep things as is. While checking, I did some clean-ups of `IAST::FormatSettings`, pushed that separately [here](https://github.com/ClickHouse/ClickHouse/pull/52332).